plugin/trino-hive/src/test/java/io/trino/plugin/hive/fs/TestTrinoFileSystemCacheStats.java

```diff
@@ -0,0 +1,81 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.fs;
+
+import io.trino.plugin.hive.s3.TrinoS3FileSystem;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.net.URI;
+
+import static org.testng.Assert.assertEquals;
+
+public class TestTrinoFileSystemCacheStats
+{
+    @Test
+    public void testCacheSizeIsCorrect()
+            throws Exception
+    {
+        TrinoFileSystemCacheStats trinoFileSystemCacheStats = new TrinoFileSystemCacheStats();
+        TrinoFileSystemCache trinoFileSystemCache = new TrinoFileSystemCache(trinoFileSystemCacheStats);
+        assertEquals(trinoFileSystemCacheStats.getCacheSize(), 0);
+        assertEquals(trinoFileSystemCache.getCacheSize(), 0);
+
+        Configuration configuration = new Configuration(true);
+        configuration.set("fs.s3.impl", TrinoS3FileSystem.class.getName());
+        trinoFileSystemCache.get(new URI("s3://bucket/path/"), configuration);
+        assertEquals(trinoFileSystemCacheStats.getGetCalls().getTotalCount(), 1);
+        assertEquals(trinoFileSystemCacheStats.getCacheSize(), 1);
+        assertEquals(trinoFileSystemCache.getCacheSize(), 1);
+
+        trinoFileSystemCache.get(new URI("s3://bucket/path1/"), configuration);
+        assertEquals(trinoFileSystemCacheStats.getGetCalls().getTotalCount(), 2);
+        assertEquals(trinoFileSystemCacheStats.getCacheSize(), 1);
+        assertEquals(trinoFileSystemCache.getCacheSize(), 1);
+
+        // use getUnique to ensure cache size is increased
+        FileSystem fileSystem = trinoFileSystemCache.getUnique(new URI("s3://bucket/path2/"), configuration);
+        assertEquals(trinoFileSystemCacheStats.getGetCalls().getTotalCount(), 2);
+        assertEquals(trinoFileSystemCacheStats.getGetUniqueCalls().getTotalCount(), 1);
+        assertEquals(trinoFileSystemCacheStats.getCacheSize(), 2);
+        assertEquals(trinoFileSystemCache.getCacheSize(), 2);
+
+        trinoFileSystemCache.remove(fileSystem);
+        assertEquals(trinoFileSystemCacheStats.getRemoveCalls().getTotalCount(), 1);
+        assertEquals(trinoFileSystemCacheStats.getCacheSize(), 1);
+        assertEquals(trinoFileSystemCache.getCacheSize(), 1);
+
+        trinoFileSystemCache.closeAll();
+        assertEquals(trinoFileSystemCacheStats.getCacheSize(), 0);
+        assertEquals(trinoFileSystemCache.getCacheSize(), 0);
+    }
+
+    @Test(expectedExceptions = IOException.class, expectedExceptionsMessageRegExp = "FileSystem max cache size has been reached: 0")
+    public void testFailedCallsCountIsCorrect()
```

Comment:
`@Test(expectedExceptions` is discouraged.
(it's acceptable to use it on a one-liner test methods, but not recommended)

use `assertThatThrownBy`.